Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DetectorFloodWeighting.py

```diff
@@ -0,0 +1,113 @@
+from mantid.api import DataProcessorAlgorithm, AlgorithmFactory, MatrixWorkspaceProperty, WorkspaceUnitValidator
+
+from mantid.kernel import Direction, FloatArrayProperty, FloatArrayBoundedValidator
+
+import numpy as np
+
+class DetectorFloodWeighting(DataProcessorAlgorithm):
+
+    def __init__(self):
+        DataProcessorAlgorithm.__init__(self)
+
+    def category(self):
+        return 'Workflow\\SANS'
+
+
+    def summary(self):
+        return 'Generates a Detector flood weighting, or sensitivity workspace'
+
+
+    def PyInit(self):
+
+        self.declareProperty(MatrixWorkspaceProperty('InputWorkspace', '',
+                                                     direction=Direction.Input, validator=WorkspaceUnitValidator("Wavelength")),
+                             doc='Flood weighting measurement')
+
+        validator = FloatArrayBoundedValidator()
+        validator.setLower(0.)
+        self.declareProperty(FloatArrayProperty('Bands', [], direction=Direction.Input, validator=validator),
+                             doc='Wavelength bands to use. Single pair min to max.')
+
+        self.declareProperty(MatrixWorkspaceProperty('OutputWorkspace', '',
+                                                     direction=Direction.Output),
+                             doc='Normalized flood weighting measurement')
+
+
+    def validateInputs(self):
+        """
+        Validates input ranges.
+        """
+        issues = dict()
+
+        bands = self.getProperty('Bands').value
+
+        if not any(bands):
+            issues['Bands'] = 'Bands must be supplied'
+
+
+        if not len(bands)%2 == 0:
+            issues['Bands'] = 'Even number of Bands boundaries expected'
+
+        if len(bands) > 2:
+            issues['Bands'] = 'Presently this algorithm only supports one pair of bands'
+
+        all_limits=list()
+        for i in range(0, len(bands), 2):
```

Comment:
You will get out of bounds here if the length of bands is odd. You should probably return at l.45